src/workerd/api/tests/js-rpc-test.js

```diff
@@ -1015,7 +1039,79 @@ export let streams = {
       await promise;
     }
 
-    // TODO(someday): Test JS-backed WritableStream, when it actually works.
+    {
+      const dec = new TextDecoder();
+      let result = '';
+      const { promise, resolve } = Promise.withResolvers();
+      const writable = new WritableStream({
+        write(chunk) {
+          result += dec.decode(chunk, {stream: true});
+        },
+        close() {
+          result += dec.decode();
+          resolve();
+        },
+      });
+      const p1 = env.MyService.writeToStream(writable);
+      await promise;
+      assert.strictEqual(result, "foo, bar, baz!");
+      await p1;
+    }
+
+    {
+      // In this test, the remote side writes a chunk to the stream below, which throws
+      // an error. Ideally the error would propagate back to the calling side so that the
+      // remote knows the stream failed and can no longer be written to. The call to
+      // writeToStreamExpectingError should throw because the error should be propagated
+      // through the round trip.
+      const dec = new TextDecoder();
+      let result = '';
+      const writable = new WritableStream({
+        write(chunk) {
+          throw new Error('boom');
+        },
+      });
+      // TODO(now): This works in one test case (js-rpc-test), but not another
+      // (js-rpc-socket-test). Need to understand why.
+      // The writeToStreamExpectingError calls write twice, The first call to write should
+      // error the stream and should cause the subsequent write to fail also. However, both
+      // writes on the remote side succeed. What we should end up with is
+      // writeToStreamExpectingError throwing an error that matches the error thrown in the
+      // write algorithm.
+      // The error thrown in the write IS being propagated up to the capnp stream created
+      // in WritableStream::serialize, but it does not get propagated back to the remote
+      // side as far as I can tell. At least, the writes happening on that side are never
+      // errored and the remote stream is not errored.
```

Comment:
In the `js-rpc-test` variant, this test case "passes", the `js-rpc-socket-test` variant, it fails, but the test log gives literally no information why it failed (there are no helpful log messages). That said, even in the `js-rpc-test` case, I don't believe the test is actually working correctly since on the remote end (in the `writeToStreamExpectingError()` method, both write awaits succeed even tho the first one should have triggerd an error. It's just not clear to me what's breaking down here.

Reply:
`js-rpc-socket-test.wd-test` had to be configured to disable info logs because otherwise the test takes 20x longer just due to symbolizing all the stack traces for logs of expected errors. This is configured in the BUILD file. You could temporarily change it while debugging. (You may also want to temporarily delete all the test cases except the one you care about.)

One reason why js-rpc-socket-test may behave differently is because when writing messages over a real socket, there is much more room for the kind of buffering I mentioned above. That is, you'll be able to queue even more write()s before one of them actually blocks.